examples/cpp/synchronous_api/main.cpp

```diff
@@ -32,41 +32,32 @@
 #include <models/input_data.h>
 #include <models/results.h>
 
+int main(int argc, char* argv[]) try {
+    if (argc != 3) {
+        throw std::runtime_error(std::string{"Usage: "} + argv[0] + " <path_to_model> <path_to_image>");
+    }
 
-int main(int argc, char* argv[]) {
-    try {
-        if (argc != 3) {
-            std::cerr << "Usage : " << argv[0] << " <path_to_model> <path_to_image>"
-                      << std::endl;
-            return EXIT_FAILURE;
-        }
-
-        cv::Mat image = cv::imread(argv[2]);
-        if (!image.data) {
-            throw std::runtime_error{"Failed to read the image"};
-        }
+    cv::Mat image = cv::imread(argv[2]);
+    if (!image.data) {
+        throw std::runtime_error{"Failed to read the image"};
+    }
 
-        // Instantiate Object Detection model
-        auto model = DetectionModel::create_model(argv[1]); // works with SSD models. Download it using Python Model API
+    // Instantiate Object Detection model
+    auto model = DetectionModel::create_model(argv[1]); // works with SSD models. Download it using Python Model API
 
-        // Run the inference
-        auto result = model->infer(image);
+    // Run the inference
+    auto result = model->infer(image);
 
-        // Process detections
-        for (auto& obj : result->objects) {
+    // Process detections
+    for (auto& obj : result->objects) {
         std::cout << " " << std::left << std::setw(9) << obj.label << " | " << std::setw(10) << obj.confidence
-                    << " | " << std::setw(4) << int(obj.x) << " | " << std::setw(4) << int(obj.y) << " | "
-                    << std::setw(4) << int(obj.x + obj.width) << " | " << std::setw(4) << int(obj.y + obj.height)
-                    << std::endl;
-        }
-
-    } catch (const std::exception& error) {
-        std::cerr << error.what() << std::endl;
-        return 1;
-    } catch (...) {
-        std::cerr << "Unknown/internal exception happened." << std::endl;
-        return 1;
+            << " | " << std::setw(4) << int(obj.x) << " | " << std::setw(4) << int(obj.y) << " | "
+            << std::setw(4) << int(obj.x + obj.width) << " | " << std::setw(4) << int(obj.y + obj.height) << "\n";
```

Comment:
Do we need `return 0;` here ?

Reply:
It turns out main is so much special that it's not required to return 0. https://en.cppreference.com/w/cpp/language/main_function